packages/pages/src/dev/server/ssr/findMatchingStaticTemplate.ts

```diff
@@ -15,12 +15,16 @@ export default async function findMatchingStaticTemplate(
       if (!isStaticTemplateConfig(t.config)) {
         return false;
       }
-      const document = await getLocalDataForEntityOrStaticPage({
-        entityId: "",
-        locale,
-        featureName: t.config.name,
-      });
-      return slug === t.getPath({ document });
+      try {
```

Comment:
Reason for change: We use this `getLocalDataForEntityOrStaticPage` to get a document to pass to the `getPath` function in order to get the path of a static template.  Since this function just iterates through all the templates using a locale that might not be relevant to other pages, it might not find a document it was asked to find and error out. For example.

We have two static pages with different locales
```
myHomePage: locales["en"]
myTacosPage: locales["en", "zh"]
```
and we are iterating through the documents, trying to find the Chinese (zh) tacos page.  At one point, `getLocalDataForEntityOrStaticPage` may attempt to run with `locale=zh` and `featureName=myHomePage` which does not exist.  This is an expected part of the iteration process, but should not break since it just means that  a document with those characteristics is not the right document.

Reply:
Great explanation! Thanks for providing this.